streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -130,7 +132,12 @@ public String topic() {
         if (task.record() == null)
             throw new IllegalStateException("This should not happen as topic() should only be called while a record is processed");
 
-        return task.record().topic();
+        String topic = task.record().topic();
```

Comment:
Explained offline -- I was missing that the returned record is actually a `StampedRecord`, not the underlying `ConsumerRecord`, so the timestamp from the dummy `ConsumerRecord` isn't actually used.